### ISSUE TITLE
Custom Parser

### ISSUE THREAD

**Issue by couturecraigj** [posthtml/express-posthtml#5]
## Problem

I like the idea of using posthtml to have a fully extensible interface for rendering my files and to remove duplication and excess white space.  What if I want to implement this in an existing project that uses another view engine parser.  I would like to say connect in pugjs to be parsed instead of re-writing the whole interface using posthtml

**Comment by michael-ciniawsky** [comment]
There is `possthtml-pug`has custom parser (Wraps Pug => PosHTML AST)

```js
posthtml(plugins).process(html, { parser: require('posthtml-pug')() })...
```

The problem is I need to update `express-posthtml`, to enable passing PostHTML Options.

**index.js**
```diff
...
if (!options.plugins && options.extend === false) {
    plugins = options.settings['view options'] || []
  } else if (options.extend === true) {
-   plugins = options.settings['view options'].concat(options.plugins)

+   plugins = options.settings['view options'].plugins
+   plugins = plugins.concat(options.plugins)
  } else {
    plugins = options.plugins || []
}
+ let options = options.settings['view options'].options || {}

  fs.readFile(path, function (err, content) {
    if (err) return cb(new Error(err))

    posthtml(plugins)
-      .process(content.toString())
+     .process(content.toString(), options) 
      .then(result => cb(null, result.html))
})
...
```

**server.js**
```diff
const pug = require('posthtml-pug')()

app.engine('html', require('posthtml'))

app.set('views', /* Path to views */)
app.set('view engine', 'html')

- app.set('view options', [])
+ app.set('view options, { plugins: [...plugins], options: { parser: pug } })
```

But i haven't time atm, maybe later that day, if you're faster send PR 😛 please

**Comment by couturecraigj** [comment]
@michael-ciniawsky 
I will see if I can do that.  The problem with the posthtml-pug is that it is not actually a view engine so in each instance I would have to call that ugly fs.readfile callback hell thing and would like my code to look a little neater and not so hacky.

**Comment by michael-ciniawsky** [comment]
It's been a while since, but if you refer to `.pug` files, the above should solve that Issue, maybe you need to set `app.set('view engine', 'pug')` instead of `app.set('view engine', 'html')` because that's where `res.render('index')` gets the file extensions from, but I'm not :100: sure tbh if anymore tbh :stuck_out_tongue:, but certain that you don't need anything else to get it working.

**Comment by michael-ciniawsky** [comment]
with `posthtml-pug` or `posthtml-sugarml`  pug/pug-like syntax is supported and gets parsed by posthtml

**Comment by michael-ciniawsky** [comment]
@couturecraigj Released in v1.1.0 👾
